article.md

```diff
@@ -0,0 +1,309 @@
+
+We will create a LinkedIn job scraper utilising Crawlee for Python to extract the company name, job title, time of posting and link to the job posting, from dynamically received user input through the web application.
+
+By the end of this tutorial, you’ll have had a fully functional web application that you can use to scrape job postings from the LinkedIn website.
+
+
+
+![alt_text](images/image1.png "image_tooltip")
+
+
+Let’s get started
+
+
+## Prerequisites
+
+
+Let's start by creating a new Crawlee for Python project with this command:
+
+
+```
+pipx run crawlee create linkedin-scraper
+```
+
+
+Select `PlaywrightCrawler` in the terminal when Crawlee asks for it.
+
+After installation, Crawlee for Python will create a boilerplate code for you. Redirect into the project folder and run this command to install dependencies.
+
+
+```
+poetry install
+```
+
+
+
+## Building the LinkedIn Scraper using Crawlee for Python
+
+In this section, we are going to be building the scraper using the Crawlee Python package. To learn more about Crawlee, check out their [documentation](https://crawlee.dev/python/docs/quick-start).
+
+
+### Inspecting the LinkedIn job Search Page
+
+Open LinkedIn on your web browser and sign out from the website (if you already have an account logged in). You should see an interface like this.
+
+
+![alt_text](images/image2.png "image_tooltip")
+
+
+Navigate to the jobs section, search for a job and location of your choice; copy the Uniform Resource Locator (URL).
+
+
+
+![alt_text](images/image3.png "image_tooltip")
+
+
+You should have something like this:
+
+`https://www.linkedin.com/jobs/search?keywords=Backend%20Developer&location=Canada&geoId=101174742&trk=public_jobs_jobs-search-bar_search-submit&position=1&pageNum=0` 
+
+
+We're going to focus on the search parameters, which is the part that goes after '?'. The keyword and location parameters are the most important ones for us. 
+
+The job title the user supplies will serve as input to the keyword parameter while the location the user supplies will go into the location parameter. Lastly, the geoId parameter will be removed while we keep the other parameters constant.
+
+We are going to be making changes to our `main.py` file. Copy and paste the code below in your `main.py` file.
+
+
+```
+from crawlee.playwright_crawler import PlaywrightCrawler
+from .routes import router                                     
+import urllib.parse
+
+async def main(title: str, location: str, data_name: str) -> None:
+    base_url = "https://www.linkedin.com/jobs/search"
+
+    # URL encode the parameters
+    params = {
+        "keywords": title,
+        "location": location,
+        "trk": "public_jobs_jobs-search-bar_search-submit",
+        "position": "1",
+        "pageNum": "0"
+    }
+    encoded_params = urllib.parse.urlencode(params)
+    encoded_url = f"{base_url}?{encoded_params}"
```

Comment:
I'm not a Pythonista, but doesn't urllib also have a function to resolve base URLs? It feels wrong to use a specialized URL manipulation library for params, and then manually stitch the strings together with the base URL.